packages/ui/src/coreModules/search/components/MultipleSearchTagsSelect/TagGroup.js

```diff
@@ -15,64 +16,76 @@ const propTypes = {
       selected: PropTypes.bool.isRequired,
     })
   ).isRequired,
+  matchingTagsReachedLimit: PropTypes.bool.isRequired,
   onDeselectAllForSearchQuery: PropTypes.func.isRequired,
   onSelectAllForSearchQuery: PropTypes.func.isRequired,
   onToggleTagSelected: PropTypes.func.isRequired,
+  searchOption: PropTypes.object.isRequired,
   searchQuery: PropTypes.string.isRequired,
-  title: PropTypes.string.isRequired,
   translateTagType: PropTypes.func.isRequired,
 }
-const TagGroup = ({
-  addTagTypeToText,
-  onDeselectAllForSearchQuery: handleDeselectAllForSearchQuery,
-  onSelectAllForSearchQuery: handleSelectAllForSearchQuery,
-  onToggleTagSelected: handleToggleTagSelected,
-  matchingTags,
-  searchQuery,
-  title,
-  translateTagType,
-}) => {
-  return (
-    <Grid verticalAlign="middle">
-      <Grid.Column width={16}>
-        <Header size="medium">
-          {title}
-          <Button
-            onClick={() => handleDeselectAllForSearchQuery(searchQuery)}
-            style={{ marginLeft: '1em' }}
-          >
-            Deselect all
-          </Button>
-          <Button onClick={() => handleSelectAllForSearchQuery(searchQuery)}>
-            Select all
-          </Button>
-        </Header>
-      </Grid.Column>
-      <Grid.Column width={16}>
-        <Label.Group>
-          {matchingTags.map(({ attributes, id, selected }) => {
-            const { tagType } = attributes
-            const tagTypeText = addTagTypeToText
-              ? ` [${translateTagType(tagType)}] `
-              : ' '
+class TagGroup extends Component {
+  componentDidMount() {
+    if (!this.props.matchingTags.length) {
+      this.props.fetchFreeTextTags(this.props.searchOption)
+    }
+  }
```

Comment:
when we start to use React hooks this change from function to class wouldn't be necessary ;-)